lib/filter_nb.c

```diff
@@ -1215,27 +1161,85 @@ lib_prefix_list_entry_ipv4_prefix_destroy(struct nb_cb_destroy_args *args)
 }
 
 /*
- * XPath: /frr-filter:lib/prefix-list/entry/ipv4-prefix-length-greater-or-equal
+ * XPath: /frr-filter:lib/prefix-list/entry/ipv4-prefix
  */
-static int lib_prefix_list_entry_ipv4_prefix_length_greater_or_equal_modify(
-	struct nb_cb_modify_args *args)
+static int
+lib_prefix_list_entry_ipv4_prefix_modify(struct nb_cb_modify_args *args)
 {
-	struct prefix_list_entry *ple;
 
-	if (args->event == NB_EV_VALIDATE &&
-	    prefix_list_length_validate(args) != NB_OK)
-		return NB_ERR_VALIDATION;
+	int af_type;
+
+	if (args->event == NB_EV_VALIDATE) {
+		const struct lyd_node *plist_dnode =
+			yang_dnode_get_parent(args->dnode, "prefix-list");
+		af_type = yang_dnode_get_enum(plist_dnode, "./type");
+		if (af_type != YPLT_IPV4) {
+			snprintf(args->errmsg, args->errmsg_len,
+				 "prefix-list type %u is mismatched.", af_type);
+			return NB_ERR_VALIDATION;
+		}
+		return NB_OK;
+	}
+
+	return lib_prefix_list_entry_prefix_modify(args);
+}
+
+static int
+lib_prefix_list_entry_ipv4_prefix_destroy(struct nb_cb_destroy_args *args)
+{
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	return lib_prefix_list_entry_prefix_destroy(args);
+}
+
+/*
+ * XPath: /frr-filter:lib/prefix-list/entry/ipv6-prefix
+ */
+static int
+lib_prefix_list_entry_ipv6_prefix_modify(struct nb_cb_modify_args *args)
+{
+
+	int af_type;
 
 	if (args->event == NB_EV_VALIDATE) {
-		if (plist_is_dup_nb(args->dnode)) {
+		const struct lyd_node *plist_dnode =
+			yang_dnode_get_parent(args->dnode, "prefix-list");
+		af_type = yang_dnode_get_enum(plist_dnode, "./type");
+		if (af_type != YPLT_IPV6) {
 			snprintf(args->errmsg, args->errmsg_len,
-				 "duplicated prefix list value: %s",
-				 yang_dnode_get_string(args->dnode, NULL));
+				 "prefix-list type %u is mismatched.", af_type);
 			return NB_ERR_VALIDATION;
 		}
 		return NB_OK;
 	}
 
+	return lib_prefix_list_entry_prefix_modify(args);
+}
+
+static int
+lib_prefix_list_entry_ipv6_prefix_destroy(struct nb_cb_destroy_args *args)
+{
+
+	if (args->event != NB_EV_APPLY)
+		return NB_OK;
+
+	return lib_prefix_list_entry_prefix_destroy(args);
+}
+
+/*
+ * XPath: /frr-filter:lib/prefix-list/entry/ipv4-prefix-length-greater-or-equal
+ */
+static int lib_prefix_list_entry_ipv4_prefix_length_greater_or_equal_modify(
+	struct nb_cb_modify_args *args)
+{
+	struct prefix_list_entry *ple;
+
+	if (args->event == NB_EV_VALIDATE &&
```

Comment:
May I know, why you are not using switch statement for handling event ?
I feel it's good If we can use switch statement here, just to be in sync how we have done in other modules.

Reply:
This file contains all NB events in non switch statement, I changed only few nb callbacks to adjust to required changes. I do intend to modify all callbacks to use switch statement. We can take that as a separate PR to convert all callbacks.